synapse/http/federation/well_known_resolver.py

```diff
@@ -125,7 +125,9 @@ async def get_well_known(self, server_name: bytes) -> WellKnownLookupResult:
         # requests for the same server in parallel?
         try:
             with Measure(self._clock, "get_well_known"):
-                result, cache_period = await self._fetch_well_known(server_name)
+                result, cache_period = await self._fetch_well_known(
+                    server_name
+                )  # type: Tuple[Optional[bytes], float]
```

Comment:
How come this is `Optional[bytes]`, whereas `_fetch_well_known` supposedly returns `bytes`?

Reply:
It is because `result` can be set to `None` below, so without it you get:

> synapse/http/federation/well_known_resolver.py:139: error: Incompatible types in assignment (expression has type "None", variable has type "bytes")  [assignment]

This isn't saying the type of the return value of `_fetch_well_known`, but the types of the variables `result` and `cache_period`.